src/components/App/App.js

```diff
@@ -1,53 +1,34 @@
-import React, { useState } from 'react';
+import React from "react";
+import { useSelector } from "react-redux";
+import "./App.css";
 
-import './App.css';
+import ErrorBoundary from "../ErrorBoundary/ErrorBoundary";
+import Footer from "../Footer/Footer";
+import Header from "../Header/Header";
+import Main from "../Main/Main";
+import MovieDetails from "../MovieDetails/MovieDetails";
 
-import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
-import Footer from '../Footer/Footer';
-import Header from '../Header/Header';
-import Main from '../Main/Main';
-import MovieDetails from '../MovieDetails/MovieDetails';
-
-import { AppContext } from '../../hooks/AppContext';
-import { useDummyHttp } from '../../hooks/useDummyHttp';
+import { useFetchMoviesData } from "../../hooks/useFetchMoviesData";
 
 const App = () => {
-	const [ isHeaderVisible, setIsHeaderVisible ] = useState(true);
-	const [ movie, setMovie ] = useState({});
-	const [ isLoading, movies, setMovies, moviesNumber, setMoviesNumber ] = useDummyHttp();
+  const isHeaderVisible = useSelector((state) => state.ui.isHeaderVisible);
+  const isLoading = useSelector((state) => state.ui.isLoading);
 
-	const appContext = {
-		moviesContext: {
-			movies,
-			setMovies,
-			moviesNumber,
-			setMoviesNumber
-		},
-		movieContext: {
-			movie,
-			setMovie
-		},
-		navigationContext: {
-			isHeaderVisible,
-			setIsHeaderVisible
-		}
-	};
+  useFetchMoviesData();
 
-	return (
-		<ErrorBoundary>
-			<AppContext.Provider value={appContext}>
-				{isHeaderVisible ? (
-					<Header />
-				) : (
-					<header>
-						<MovieDetails />
-					</header>
-				)}
-				<Main isLoading={isLoading} />
-				<Footer />
-			</AppContext.Provider>
-		</ErrorBoundary>
-	);
+  return (
+    <ErrorBoundary>
+      {isHeaderVisible ? (
+        <Header />
```

Comment:
It seems like we have two headers here. Better to move `<header>
          <MovieDetails />
        </header>` to Header component

Reply:
All done.